layer/support/event_logging.h

```diff
@@ -169,6 +183,23 @@ class TraceEventAttr : public Attribute {
 
   const std::vector<Attribute *> &GetArgs() const { return args_; }
 
+  template <typename AttrType>
+  const AttrType *GetArg() const {
+    for (Attribute *arg : args_) {
+      if (auto *attr = arg->cast<AttrType>()) return attr;
+    }
+    return nullptr;
+  }
+
+  template <typename AttrType>
+  const AttrType *GetArg(std::string_view name) const {
+    for (Attribute *arg : args_) {
+      if (auto *attr = arg->cast<AttrType>(); attr->GetName() == name)
```

Comment:
ah, this also needs a null check `attr && attr->GetName() == name)`
Sorry, I forgot about this earlier

Reply:
or alternatively you can decompose this into 2 ifs:
```c++
if (auto *attr = arg->cast<AttrType>())
  if (attr->GetName() == name)
    return attr;
```